CMakeLists.txt

```
@@ -0,0 +1,58 @@
+cmake_minimum_required(VERSION 3.4)
+project(microprofile)
+
+option(MICROPROFILE_USE_CONFIG_FILE "Use user provided configuration in microprofile.config.h file." OFF)
+
+set(MICROPROFILE_EXPORT_FILENAME microprofile.export.h)
+set(MICROPROFILE_CONFIG_HEADER ${PROJECT_SOURCE_DIR}/microprofile.config.h)
+set(MICROPROFILE_PUBLIC_HEADERS
+        ${PROJECT_SOURCE_DIR}/microprofile.h
+        ${CMAKE_CURRENT_BINARY_DIR}/${MICROPROFILE_EXPORT_FILENAME}
+)
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+
+add_library(${PROJECT_NAME} microprofile.cpp)
+
+include(GenerateExportHeader)
+generate_export_header(${PROJECT_NAME}
+        EXPORT_MACRO_NAME MICROPROFILE_API
+        EXPORT_FILE_NAME ${MICROPROFILE_EXPORT_FILENAME}
+)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+        $<INSTALL_INTERFACE:include>
```

`CMAKE_INSTALL_INCLUDEDIR` is usually a relative path.
`CMAKE_INSTALL_FULL_INCLUDEDIR` is the fulll path.
See https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html#gnuinstalldirs
```suggestion
        $<INSTALL_INTERFACE:${MAKE_INSTALL_INCLUDEDIR}>
```